custom_components/miele/sensor.py

```diff
@@ -540,6 +552,60 @@ def device_state_attributes(self):
         return attributes
 
 
+class MieleEnergyConsumptionSensor(SensorEntity):
+    def __init__(self, hass, device, key):
+        self._hass = hass
+        self._device = device
+        self._key = key
+
+        self._attr_device_class = DEVICE_CLASS_ENERGY
+        self._attr_state_class = STATE_CLASS_TOTAL_INCREASING
+        self._attr_native_unit_of_measurement = 'kWh'
+
+    @property
+    def device_id(self):
+        """Return the unique ID for this sensor."""
+        return self._device["ident"]["deviceIdentLabel"]["fabNumber"]
+
+    @property
+    def unique_id(self):
+        """Return the unique ID for this sensor."""
+        return self.device_id + "_" + self._key
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        ident = self._device["ident"]
+
+        result = ident["deviceName"]
+        if len(result) == 0:
+            return ident["type"]["value_localized"] + " " + _map_key(self._key)
+        else:
+            return result + " " + _map_key(self._key)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        device_state = self._device["state"]
+
+        if "ecoFeedback" in device_state and device_state["ecoFeedback"] is not None:
+            if "currentEnergyConsumption" in device_state["ecoFeedback"]:
+                consumption = device_state["ecoFeedback"]["currentEnergyConsumption"]
+
+                if consumption["unit"] == 'kWh':
+                    return consumption["value"]
+                elif consumption["unit"] == 'Wh':
+                    return consumption["value"] / 1000.0
+
+        return 0
+
+    async def async_update(self):
+        if not self.device_id in self._hass.data[MIELE_DOMAIN][DATA_DEVICES]:
+            _LOGGER.debug("Miele device disappeared: {}".format(self.device_id))
+        else:
+            self._device = self._hass.data[MIELE_DOMAIN][DATA_DEVICES][self.device_id]
```

Comment:
There's a bit of duplication going on here, but I don't really know what to do about it. This is more or less the first time I'm working with a Home Assistant integration, so bear with me 😄